TfIdf_BOW_creator.py

```diff
@@ -0,0 +1,186 @@
+        word = re.sub(r'[^a-z\s]','',word)
+        word = ps.stem(word)
+        if len(word)>2 and dict.check(word):
+            processed_word = word
+        else:
+            processed_word = ""
+
+        processed_sentence+=" "+processed_word
+    return processed_sentence
+
+# Create vocab
+def create_vocab(filename):
+    # open the file in read mode
+    read_file = open(filename, 'r')
+    
+    # creating dictreader object
+    file = csv.DictReader(read_file)
+    
+    # creating empty lists
+    current_sentence = list()
+    is_oligo = list()
+    
+    # iterating over each row and append
+    # values to empty list
+    for col in file:
+        current_sentence.append(col['Current Sentence'])
+        is_oligo.append(col['Is this an oligo?'])
+
+    non_oligo_sentences = list()
+    oligo_sentences = list()
+    non_oligo_vocab = list()
+    oligo_vocab = list()
+
+    for i in range(len(current_sentence)):
+        if is_oligo[i]=="Oligo" and current_sentence[i] not in oligo_sentences: #appends unique sentences only
+            processed_sentence = sentence_processor(current_sentence[i])
+            # oligo_sentences.append(current_sentence[i])
+            oligo_sentences.append(processed_sentence)
+
+            for word in processed_sentence.split():
+                if word not in oligo_vocab:
+                    processed_word = word_processor(word)
+                    if not processed_word: #if not oligo then append that in vocab coz we want that are linked
+                        # with oligo in a sentence, but are themselves not an oligo
+                        oligo_vocab.append(word)
+                        # processed_bow_word = bow_processor(word)
+                        # if not processed_bow_word:
+                        #     oligo_vocab.append(processed_bow_word) 
+
+
+        elif is_oligo[i]!="Oligo" and current_sentence[i] not in non_oligo_sentences:
+            processed_sentence = sentence_processor(current_sentence[i])
+            # oligo_sentences.append(current_sentence[i])
+            non_oligo_sentences.append(processed_sentence)
+
+            for word in processed_sentence.split():
+                if word not in non_oligo_vocab:
+                    processed_word = word_processor(word)
+                    if not processed_word: #if not oligo looking sequence then append that in vocab coz 
+                        # we want that are linked with oligo in a sentence, but are themselves not an oligo
+                        non_oligo_vocab.append(word)
+
+    return non_oligo_sentences, oligo_sentences, non_oligo_vocab, oligo_vocab
+
+
+# Creating an index for each word in our vocab.
+def vocab_elements(vocab):
+    index_dict = {} 
+    i = 0
+    for word in vocab:
+        index_dict[word] = i
+        i += 1
+    return index_dict
+
+
+# Create a count dictionary to keep the count of the number of documents containing the given word
+def count_dict(sentences, vocab):
```

Comment:
Python also provides a Counter class for this kind of tasks:
https://docs.python.org/3/library/collections.html#collections.Counter

Reply:
"count_dict" is just the name of the function, it's not related to the "counter" class.